--- /usr/share/X11/xkb/symbols/be	2019-07-24 19:14:15.371046157 +0200
+++ /usr/share/X11/xkb/symbols/be	2019-07-24 19:16:12.007525230 +0200
@@ -4,6 +4,7 @@
 xkb_symbols "basic" {
 
     include "latin"
+    include "mswindows-capslock"
 
     name[Group1]="Belgian";

--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ /usr/share/X11/xkb/symbols/mswindows-capslock	2019-07-24 19:16:51.312960829 +0200
@@ -0,0 +1,19 @@
+// $XKeyboardConfig$
+
+// Replicate a "feature" of MS Windows on AZERTY keyboards 
+// where Caps Lock also acts as a Shift Lock on number keys.
+// Include keys <AE01> to <AE10> in the FOUR_LEVEL_ALPHABETIC key type.
+
+partial alphanumeric_keys
+xkb_symbols "basic" {
+    key <AE01>	{ type= "FOUR_LEVEL_ALPHABETIC", [ ampersand,          1,          bar,   exclamdown ]	};
+    key <AE02>	{ type= "FOUR_LEVEL_ALPHABETIC", [    eacute,          2,           at,    oneeighth ]	};
+    key <AE03>	{ type= "FOUR_LEVEL_ALPHABETIC", [  quotedbl,          3,   numbersign,     sterling ]	};
+    key <AE04>	{ type= "FOUR_LEVEL_ALPHABETIC", [apostrophe,          4,   onequarter,       dollar ]	};
+    key <AE05>	{ type= "FOUR_LEVEL_ALPHABETIC", [ parenleft,          5,      onehalf, threeeighths ]	};
+    key <AE06>	{ type= "FOUR_LEVEL_ALPHABETIC", [   section,          6,  asciicircum,  fiveeighths ]	};
+    key <AE07>	{ type= "FOUR_LEVEL_ALPHABETIC", [    egrave,          7,    braceleft, seveneighths ]	};
+    key <AE08>	{ type= "FOUR_LEVEL_ALPHABETIC", [    exclam,          8,  bracketleft,    trademark ]	};
+    key <AE09>	{ type= "FOUR_LEVEL_ALPHABETIC", [  ccedilla,          9,    braceleft,    plusminus ]	};
+    key <AE10>	{ type= "FOUR_LEVEL_ALPHABETIC", [    agrave,          0,   braceright,       degree ]	};
+};
